### PR TITLE
Fix the probability formula for some topic given some words

### PR DESCRIPTION
**Pull request by JiaxiangBU**
Hi, @ethen8181 I read your good intro for Gibbs Sampling. And I find a typo in one latex formula.

https://github.com/ethen8181/machine-learning/blob/1f71423da54bfde24de7528a3ef0f5c9e694f4b7/clustering_old/topic_model/LDA.Rmd#L149-L152

https://github.com/ethen8181/machine-learning/blob/1f71423da54bfde24de7528a3ef0f5c9e694f4b7/clustering_old/topic_model/LDA.Rmd#L198

Here is the thing, for the first iteration, you script does a random assignment for the first word in the first document. Here ` left * right` with two values, the probability of topic 1 and that of topic 2.
The sum of both is not equal to 1. It just displays the relative weight for this word between two topics. Thus, they need do a normalization.

I double-check [the reference] you list at the end of the article.
The author does use `\propto` to show relative weights for topics.

[the reference]: https://www.cl.cam.ac.uk/teaching/1213/L101/clark_lectures/lect7.pdf

Thus, I open a PR for further discussion.

### CODE REVIEW

**Comment by ethen8181** [comment]
Thanks for the feedback.

1. Regarding the latex formula, sure, changing it to proportional instead of equal makes sense.
2. For the sample weight, from an implementation standpoint, does it matter whether we normalize it? R's sample accepts a probability weight that is not normalized?

File: clustering_old/topic_model/LDA.Rmd [hunk 1]
```diff
@@ -194,8 +194,12 @@ wt[t0, wid] <- wt[t0, wid] - 1
 left  <- ( wt[, wid] + eta ) / ( rowSums(wt) + length(vocab) * eta )
 right <- ( dt[1, ] + alpha ) / ( sum( dt[1, ] ) + K * alpha )
 
+probs <- left * right
+probs <- probs/sum(probs)
+
 # draw new topic for the first word in the first document 
-t1 <- sample(1:K, 1, prob = left * right)
+t1 <- sample(1:K, 1, prob = probs)
```
Comment (JiaxiangBU) [inline clustering_old/topic_model/LDA.Rmd]: I agree with you, add a note for `sample` like 

```suggestion
# The optional prob argument can be used to give a vector of weights for obtaining the elements of the vector being sampled. They need not sum to one, but they should be non-negative and not all zero.
t1 <- sample(1:K, 1, prob = probs)
```

**Review by JiaxiangBU — COMMENTED** [review]
@ethen8181 

> 2. For the sample weight, from an implementation standpoint, does it matter whether we normalize it? R's sample accepts a probability weight that is not normalized?

Yes, I check the help document, in the detail part, there is an explanation.

> The optional prob argument can be used to give a vector of weights for obtaining the elements of the vector being sampled. They need not sum to one, but they should be non-negative and not all zero.

**Comment by ethen8181** [comment]
nitpick:

1. can you add back the space between left and right, so `left * right` instead of `left*right`.
2. It would be great if you could also generate the .HTML doc from the Rmarkdown file.

I can merge this once you resolve these two comments. Thanks.

**Comment by JiaxiangBU** [comment]
and notes and space for relative probability weight `left * right`, render `LDA.Rmd` and use the relative path for the script`LDA_functions.R` for reproducible purpose. @ethen8181 see 2dc23e2

**Comment by ethen8181** [comment]
Thanks. merged.

**Pull request merged**